ci/infra/openstack/generate-cloud-provider-conf

```diff
@@ -0,0 +1,91 @@
+#!/bin/bash
+
+umask 0077
+
+cat << EOF > openstack.conf
+[Global]
+auth-url="${OS_AUTH_URL}"
+username="${OS_USERNAME}"
+password="${OS_PASSWORD}"
+tenant-name=
+tenant-id=
+domain-name=
+domain-id=
+region="${OS_REGION_NAME}"
+ca-file=
+[LoadBalancer]
+lb-version=v2
+subnet-id="${OS_PRIVATE_SUBNET_ID}"
+floating-network-id="${OS_PUBLIC_NET_ID}"
+create-monitor=yes
+monitor-delay=1m
+monitor-timeout=30s
+monitor-max-retries=3
+[BlockStorage]
+trust-device-path=false
+bs-version=v2
+ignore-volume-az=true
+EOF
+
+umask 0022
+
+if [ -n "$OS_PROJECT_ID" ]; then
+  sed -i '/^tenant-name=/d
+          /^tenant-id=/ s|=.*|="'${OS_PROJECT_ID}'"|
+' openstack.conf
+elif [ -n "$OS_PROJECT_NAME" ]; then
+  sed -i '/^tenant-id=/d
+          /^tenant-name=/ s|=.*|="'${OS_PROJECT_NAME}'"|
+' openstack.conf
+else
+  echo "[ WARN ] \$OS_PROJECT_NAME and \$OS_PROJECT_ID weren't provided"
+fi
+
+if [ -n "$OS_USER_DOMAIN_ID" ]; then
+  sed -i '/^domain-name=/d
+          /^domain-id=/ s|=.*|="'${OS_USER_DOMAIN_ID}'"|
+' openstack.conf
+elif [ -n "$OS_USER_DOMAIN_NAME" ]; then
+  sed -i '/^domain-id=/d
+          /^domain-name=/ s|=.*|="'${OS_USER_DOMAIN_NAME}'"|
+' openstack.conf
+else
+  echo "[ WARN ] \$OS_USER_DOMAIN_NAME and \$OS_USER_DOMAIN_ID weren't provided"
+fi
+
+if [ -n "$CA_FILE" ]; then
+  sed -i '/^ca-file=/ s|=.*|="'${CA_FILE}'"|
+' openstack.conf
+else
+  sed -i '/^ca-file=/d' openstack.conf
+fi
+
+echo; echo "[ INFO ] ### Run following commands to bootstrap skuba cluster:"; echo
+
+echo " skuba cluster init --control-plane ${TR_LB_IP} --cloud-provider openstack ${TR_STACK}-cluster"
+echo " mv openstack.conf ${TR_STACK}-cluster/cloud/openstack.conf"
+echo " cd ${TR_STACK}-cluster"
+
+if [ -n "$TR_MASTER_IPS" ]; then
+  MASTERS=($TR_MASTER_IPS)
+  echo " skuba node bootstrap --target ${MASTERS[0]} --sudo --user sles caasp-master-${TR_STACK}-0"
+  if [ "${#MASTERS[@]}" -gt "1" ]; then
+    i=0
+    for MASTER in $TR_MASTER_IPS; do
+      if [ $i -eq "0" ]; then ((++i)); continue; fi
+      echo " skuba node join --role master --target $MASTER --sudo --user sles caasp-master-${TR_STACK}-${i}"
+    done
+  fi
+else
+  echo; echo "[ WARN ] \$TR_MASTER_IPS wasn't provided"
+fi
+
+if [ -n "$TR_WORKER_IPS" ]; then
+  i=0
+  for WORKER in $TR_WORKER_IPS; do
+    echo " skuba node join --role worker --target $WORKER --sudo --user sles caasp-worker-${TR_STACK}-${i}"
+    ((++i))
```

Comment:
As above, the `${i}` in the `echo` could be replaced with `$((i++))` and remove the need for the separate arithmatic expansion.  And initialize `i` to `-1` to start at `0`, since `++i` and `i++` both expand to the result after incrementing. :/

Reply:
I would like to keep it as it is, since it is easy to read and understand